### DISCUSSION TITLE
Twitter Scraper

### DISCUSSION THREAD

**Discussion by mlkorra** [tattle-made/memebox#11] [General]
Using the [tweepy](https://github.com/tweepy/tweepy) library for scraping twitter using the given tweet url

For one of the [sample](https://twitter.com/Sci_Phile/status/1457061047513141248) tweets,which has video in the tweet,the API recognizes and downloads it as image/jpg file.

This is the output after running the script,
`(memebox) mlkorra@dev:~/Dev/Tattle/tattle-made/memebox/services/processors$ python twitter_scraper.py 
{'hashtags': [], 'symbols': [], 'user_mentions': [], 'urls': [], 'media': [{'id': 1457061026420039683, 'id_str': '1457061026420039683', 'indices': [34, 57], 'media_url': 'http://pbs.twimg.com/ext_tw_video_thumb/1457061026420039683/pu/img/Hqiwp4XidZ3xJT2i.jpg', 'media_url_https': 'https://pbs.twimg.com/ext_tw_video_thumb/1457061026420039683/pu/img/Hqiwp4XidZ3xJT2i.jpg', 'url': 'https://t.co/nADw7ICRiv', 'display_url': 'pic.twitter.com/nADw7ICRiv', 'expanded_url': 'https://twitter.com/Sci_Phile/status/1457061047513141248/video/1', 'type': 'photo', 'sizes': {'thumb': {'w': 150, 'h': 150, 'resize': 'crop'}, 'small': {'w': 680, 'h': 680, 'resize': 'fit'}, 'medium': {'w': 720, 'h': 720, 'resize': 'fit'}, 'large': {'w': 720, 'h': 720, 'resize': 'fit'}}}]}
`

Also,explored other libraries,but they don't support extracting tweets/media from tweet given a tweet url

**Comment by dennyabrain** [comment]
@mlkorra then lets just focus on downloading tweet text.
And lets see if we can send an apt error like unsupported media type for tweets with media items

**Reply by mlkorra** [reply]
Okay,we were able to download the images but the videos could not

**Reply by dennyabrain** [reply]
oh i see. got it. that not too bad. lets do text tweet and images then.
